### ISSUE TITLE
Specify that warnings should give an exitcode != 0

### ISSUE THREAD

**Issue by thernstig** [microsoft/pyright#1651]
**Is your feature request related to a problem? Please describe.**
I do not want Pyright to give me any warnings, and I want Pyright warnings to give an exit code !=0 when running in CI, to fail my lint steps.

**Describe the solution you'd like**
I want my CI jobs to fail when there is an error reported. For example:

```sh
145:37 - warning: "v" is possibly unbound (reportUnboundVariable)
0 errors, 1 warnings, 0 infos
Completed in 0.559sec
~/code/project (master)> # this has no exit code in it
```

**Additional context**
ESLint has a mechanism to achieve this with `--max-warnings=0`. Maybe a similar flag to the `pyright` CLI and more importantly a config option in `pyrightconfig.json` should exist.

**Comment by erictraut** [comment]
It is intentional that warnings to not generate a non-zero error code. If you want to receive a non-zero error code in this case, you should configure pyright to emit an error for that class of diagnostic. To do this, update your pyrightconfig.json file as follows:

```
{
  "reportUnboundVariable": "error"
}
```

**Comment by KevOrr** [comment]
@erictraut it definitely makes sense to me that the _default_ behavior should be to treat warnings as non fatal, but would it not make sense to have a default-off flag to treat warnings as errors? In a CI status check, it's very useful to be able to distinguish between "passed with warnings" and "failed". Having such a flag would be a great alternative to having to either a) keeping two pyright configs which must be kept in sync, or b) relying on a somewhat hacky search-and replace step before pyright is invoked (e.g. `sed -Ei 's/"warning"/"error"/g' pyrightconfig.json`)

I'd be happy to PR this

**Comment by erictraut** [comment]
We generally add new command-line options only when we're convinced that there's sufficient justification for a feature. Each new command-line option has a cost in terms of code complexity, documentation and maintenance. I'm still not personally convinced this meets that bar, but you make a reasonable point above.

My colleagues and I will discuss and get back to you.

**Comment by erictraut** [comment]
I discussed this with my colleagues. We've decided to implement this feature.
